### DISCUSSION TITLE
How to join two input.harbor ?

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#3969] [General]
Hello there, how I can join two or more input.harbor for example to multiple DJs at same time?

```
dj1 = amplify(override="replay_gain", streaming_gain_dj,input.harbor(id="aovivo", icy=true,"/",port=8080,auth=dj_auth, replay_metadata=true, icy_metadata_charset = "UTF-8", metadata_charset = "UTF-8", on_connect=fun (_) -> dj_conectado("on",porta), on_disconnect=fun () -> dj_desconectado("off",porta), buffer = 20.00, max = 25.00))

dj2 = amplify(override="replay_gain", streaming_gain_dj,input.harbor(id="aovivo", icy=true,"/",port=8082,auth=dj_auth, replay_metadata=true, icy_metadata_charset = "UTF-8", metadata_charset = "UTF-8", on_connect=fun (_) -> dj_conectado("on",porta), on_disconnect=fun () -> dj_desconectado("off",porta), buffer = 20.00, max = 25.00))
```
Thanks in advance!

**Comment by vitoyucepi** [comment]
Hi @caf2022,
What do you mean by
> join two or more input.harbor at same time

**Reply by caf2022** [reply]
For two DJ broadcast at the same time, for example one on USA and other on Brazil everyone connect on a port(input.harbor) and then the LS merge que two sources/audio in one.

**Reply by vitoyucepi** [reply]
You are probably looking for the [`add`](https://www.liquidsoap.info/doc-2.2.4/reference.html#add) function, but I don't know how you would solve the synchronization problem.

**Comment by Russsgithub** [comment]
@caf2022 

This may help with sync

https://vdo.ninja/

I was looking at it a while ago. If I remember rightly you can create a webrtc video chat room and it provides a link for the room which can be used in OBS. It can also be used as an input into liquidsoap ( I'm pretty sure I tested this ) , you could then stream the video and audio or drop the video and just stream the audio. The djs would also be able to listen to each other in a realtime type way.

It may not work , but thought it was worth sharing .

Good luck
